File: CS/DXExample.Module/DomainObject1.cs

using DevExpress.Xpo;

using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace DXExample.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private string _property1;
        public string Property1 {
            get { return _property1; }
            set { SetPropertyValue("Property1", ref _property1, value); }
        }
        private string _property2;
        public string Property2 {
            get { return _property2; }
            set { SetPropertyValue("Property2", ref _property2, value); }
        }
        private string _property3;
        public string Property3 {
            get { return _property3; }
            set { SetPropertyValue("Property3", ref _property3, value); }
        }
        private string _property4;
        public string Property4 {
            get { return _property4; }
            set { SetPropertyValue("Property4", ref _property4, value); }
        }
        private string _property5;
        public string Property5 {
            get { return _property5; }
            set { SetPropertyValue("Property5", ref _property5, value); }
        }
    }
}

File: VB/DXExample.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace DXExample.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _property1 As String
		Public Property Property1() As String
			Get
				Return _property1
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Property1", _property1, value)
			End Set
		End Property
		Private _property2 As String
		Public Property Property2() As String
			Get
				Return _property2
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Property2", _property2, value)
			End Set
		End Property
		Private _property3 As String
		Public Property Property3() As String
			Get
				Return _property3
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Property3", _property3, value)
			End Set
		End Property
		Private _property4 As String
		Public Property Property4() As String
			Get
				Return _property4
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Property4", _property4, value)
			End Set
		End Property
		Private _property5 As String
		Public Property Property5() As String
			Get
				Return _property5
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Property5", _property5, value)
			End Set
		End Property
	End Class
End Namespace
